src/hashtable.py

```diff
@@ -51,8 +51,26 @@ def insert(self, key, value):
 
         Fill this in.
         '''
-        pass
-
+        #Overwrite keys when same key entered
+        if self.retrieve(key):
+            self.remove(key)
+        #Acquire index
+        index = self._hash_mod(key)
+        #Acqure the given Node
+        node = self.storage[index]
+
+        #If the node is empty at given index, add it and exit function
+        if node is None:
+            self.storage[index] = LinkedPair(key,value)
+            return
+        #Otherwise collision is dectected at given index
+        #Iterate through the LL
+        prev = node
+        while node is not None:
```

Comment:
Excellent implementation and commented code. Good use of class methods `retrieve()` and `remove()`.